ignite/pkg/cosmosgen/templates/composable-root/useClient.ts.tpl

```diff
@@ -0,0 +1,15 @@
+import { Client } from '{{ .PackageNS }}-client-ts'
+import { env } from '../env';
+
+const useClientInstance = () => {
+  const client = new Client(env);
+  return client;
+};
+let clientInstance: ReturnType<typeof useClientInstance>;
+
+export const useClient = () => {
+  if (!clientInstance) {
+    clientInstance = useClientInstance();
+  }
+  return clientInstance;
+};
```

Comment:
Is the local `useClientInstance` really needed? It looks like it could be simpler to create the `Client` inside the exported function instead.

Reply:
> We used this pattern heavily in emeris for performance reasons. Youll notice that the clientInstance is declared at the top-level of the file which is only ever interpreted once when the app runs effectively turning this into a singleton. As useClient() will be the most often-called hook by various other hooks and components it avoids generating multiple instances of the client.